base/mpfr.jl

```diff
@@ -706,6 +706,15 @@ function min(x::BigFloat, y::BigFloat)
     return z
 end
 
+function Base._extrema_rf(x::NTuple{2,BigFloat}, y::NTuple{2,BigFloat})
+    (x1, x2), (y1, y2) = x, y
+    isnan(x1) && return x
+    isnan(y1) && return y
+    z1 = x1 < y1 || signbit(x1) > signbit(y1) ? x1 : y1
+    z2 = x2 < y2 || signbit(x2) > signbit(y2) ? y2 : x2
+    z1, z2
+end
```

Comment:
Is it for "undoing" the branch-free optimization for `AbstractFloat`?

https://github.com/JuliaLang/julia/blob/ece2a096617e2b016e194c28bcc46891959eb20b/base/reduce.jl#L785-L793

If so, how about just calling the generic method `invoke(_extrema_rf, Tuple{Any,Any}, x, y)`? If the computation of `z1` and `z2` is correct, why not do this in `min` and `max` as well? Also, it's better to put some comments on why certain optimization is done and why it is correct.

I'm not familiar with `BigFloat` so I don't know if it's OK. It may be better to open a separate PR that focuses on optimizing `min`/`max`.

Reply:
See #41709, `BigFloat`'s `min/max` do allocation for non-nan inputs.
But `extrema` for `BigFloat` don't create new `BigFloat` before this PR.
More allocations might be considered as a regression, so I add it in `mpfr.jl` (for  bootstrap)

Reply:
> Do you mind actually merging the branch of #36265 (https://github.com/tkf/julia/tree/extrema)?
 
I dont mind, the easizer way for me might be cherry pick the commits from #36265.
But I got lost when I look into git histry... (So I choosed to copy the code directly...)

Reply:
Comments added.

Reply:
The right way to fix this seems to make `min`/`max` non-allocating. It's better to clarify the desired semantics of `min`/`max` first. If it really needs to allocate (I doubt it), we can't ignore the semantics of `min`/`max` in `extrema`.

Reply:
Sounds reasonable to me, we can put related change in #41709.